src/storage-blob-preview/azext_storage_blob_preview/operations/blob.py

```diff
@@ -586,6 +586,22 @@ def snapshot_blob(client, metadata=None, **kwargs):
     return client.get_blob_properties()
 
 
+# pylint: disable=protected-access
+def _adjust_block_blob_size(client, blob_type, length):
+    if not blob_type or blob_type != 'block':
+        return
+    # increase the block size to 100MB when the block list will contain more than
+    # 50,000 blocks(each block 4MB)
+    if length > 50000 * 4 * 1024 * 1024:
+        client._config.max_block_size = 100 * 1024 * 1024
+        client._config.max_single_put_size = 256 * 1024 * 1024
+    # increase the block size to 4000MB when the block list will contain more than
+    # 50,000 blocks(each block 100MB)
+    if length > 50000 * 100 * 1024 * 1024:
```

Comment:
with two `if`, if the length > 50000 * 100* 1024 * 1024, `client_config` will be change twice. Could we use `if ... elif ...`?

Reply:
You are right, the value will be overwrote twice. I just want the code flow more close to human mind. When 5000 blocks of 4MB is not enough, we should increase the block size to 100MB. When 100MB is not enough again, we should increase to 4000MB.

Reply:
Because we already have the capability to PUT with block size `4000 * 1024 * 1024`, could we just remove the first `if`?

Reply:
Do you mean for all `length > 50000 * 4 * 1024 * 1024`, we set the `_config.max_block_size` to 4000MB, just skip 100MB? I think it's workable.

Reply:
Made the change.